docassemble/ALAutomatedTestingTests/al_set_up_testing.py

```diff
@@ -138,6 +164,39 @@ def make_pull_request( self ):
     
     return self
   
+  def create_secrets( self ):
+    """Set the GitHub repo secrets the tests need to log into the da server and
+    create projects to contain the interviews being tested. PyGithub does not yet handle secrets. See issue: https://github.com/PyGithub/PyGithub/issues/1373."""
+    self.put_secret( 'PLAYGROUND_EMAIL', self.email )
+    self.put_secret( 'PLAYGROUND_PASSWORD', self.password )
+    self.put_secret( 'PLAYGROUND_ID', self.playground_id )
+    return self
+  
+  def put_secret( self, name, value ):
+    """Add or update one secret to the GitHub repo."""
+    # Create repo secret: https://docs.github.com/en/rest/reference/actions#create-or-update-a-repository-secret
+    # Convert the message and key to Uint8Array's (Buffer implements that interface)
+    encrypted_key = public.PublicKey( self.public_key.encode("utf-8"), encoding.Base64Encoder() )
+    sealed_box = public.SealedBox( encrypted_key )  # ?
+    encrypted = sealed_box.encrypt( value.encode( "utf-8" ))  # LibSodium
+    base64_encrypted = b64encode( encrypted ).decode( "utf-8" )  # turns into string
+    
+    url = self.github_repo_base + "/actions/secrets/" + name
+    payload = '{"encrypted_value":"' + base64_encrypted + '", "key_id":"' + self.key_id + '"}'
+    headers = {
+      'Accept': "application/vnd.github.v3+json",
+      'Authorization': self.basic_auth,
+    }
+    
+    secret_put = requests.request( "PUT", url, data=payload, headers=headers )
+    # Cannot check the value, but can check it exists
+    secret_get = requests.request( "GET", url, data="", headers=headers )
```

Comment:
Do we need to check anything about this `secret_get` variable, or is it enough that it gets set without any errors?

Reply:
I'm not even yet sure if it'll cause errors. I think it's partly aspirational. We can get rid of it if it causes us too much pain or guilt. Or comment it out. Would love to keep a record of it somewhere until we can look into it, but that could be in an issue.

Reply:
I'm hoping to include it in the error-handling work, but the details are burning me out a bit.